pipeline/src/stream_id_string.rs

```diff
@@ -0,0 +1,378 @@
+    ) -> datafusion::common::Result<GenericByteArray<GenericStringType<i32>>> {
+        let mut strs =
+            StringBuilder::with_capacity(number_rows, STREAM_ID_STRING_BYTES * number_rows);
+        for stream_id in stream_ids {
+            if let Some(stream_id) = stream_id {
+                strs.append_value(
+                    StreamId::try_from(stream_id)
+                        .map_err(|err| exec_datafusion_err!("Error {err}"))?
+                        .to_string(),
+                );
+            } else {
+                strs.append_null()
+            }
+        }
+        Ok(strs.finish())
+    }
+}
+
+impl ScalarUDFImpl for StreamIdString {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+    fn name(&self) -> &str {
+        "stream_id_string"
+    }
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+    fn return_type(&self, args: &[DataType]) -> datafusion::common::Result<DataType> {
+        if let DataType::Dictionary(key, _value) = &args[0] {
+            Ok(DataType::Dictionary(
+                key.to_owned(),
+                Box::new(DataType::Utf8),
+            ))
+        } else {
+            Ok(DataType::Utf8)
+        }
+    }
+    fn invoke_batch(
+        &self,
+        args: &[ColumnarValue],
+        number_rows: usize,
+    ) -> datafusion::common::Result<ColumnarValue> {
+        let args = ColumnarValue::values_to_arrays(args)?;
+
+        if let Some(dict) = args[0].as_any_dictionary_opt() {
+            let stream_ids = as_binary_array(dict.values())?;
+            let ids = self.map_ids(stream_ids, number_rows)?;
+            Ok(ColumnarValue::Array(dict.with_values(Arc::new(ids))))
+        } else {
+            let stream_ids = as_binary_array(&args[0])?;
+            let ids = self.map_ids(stream_ids, number_rows)?;
+            Ok(ColumnarValue::Array(Arc::new(ids)))
+        }
+    }
+}
+
+/// ScalarUDF to convert a list of binary StreamIds into a list of strings for easier inspection.
+#[derive(Debug)]
+pub struct StreamIdStringList {
+    signature: Signature,
+}
+
+impl Default for StreamIdStringList {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl StreamIdStringList {
+    /// Construct new instance
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::new(
+                TypeSignature::Exact(vec![DataType::new_list(DataType::Binary, true)]),
+                Volatility::Immutable,
+            ),
+        }
+    }
+}
+
+impl ScalarUDFImpl for StreamIdStringList {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+    fn name(&self) -> &str {
+        "array_stream_id_string"
+    }
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+    fn return_type(&self, _args: &[DataType]) -> datafusion::common::Result<DataType> {
+        Ok(DataType::new_list(DataType::Utf8, true))
+    }
+    fn invoke(&self, args: &[ColumnarValue]) -> datafusion::common::Result<ColumnarValue> {
```

Comment:
probably want to update this to use `invoke_batch` too?

Reply:
ah yep, thanks. So many of these floating around I missed this one.I'll get it updated